### PR TITLE
feat: add linting and building workflows

### PR DESCRIPTION
**Pull request by ArnabXD**

### CODE REVIEW

File: .github/workflows/typescript-build.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: npm ci
+      - name: TypeScript Build Test
+        run: npm run build --noEmit
```
Comment (roj1512) [inline .github/workflows/typescript-build.yml:26]: There would be no problem with emitting.

```suggestion
        run: npm run build
```

**Review by roj1512 — CHANGES_REQUESTED** [review]

File: .github/workflows/typescript-build.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: TypeScript Build
```
Comment (roj1512) [inline .github/workflows/typescript-build.yml:1]:
```suggestion
name: Build
```

File: .github/workflows/typescript-build.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        node-version: [16.x]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install dependencies
+        run: npm ci
+      - name: TypeScript Build Test
```
Comment (roj1512) [inline .github/workflows/typescript-build.yml:25]:
```suggestion
      - name: Build
```

File: .github/workflows/eslint.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: eslint
```
Comment (roj1512) [inline .github/workflows/eslint.yml:1]:
```suggestion
name: Lint
```

**Review by roj1512 — CHANGES_REQUESTED** [review]
I also suggest renaming `eslint.yml` to `lint.yml` and `typescript-build.yml` to `build.yml`.

**Comment by ArnabXD** [comment]
BTW do checkout [this issue](https://github.com/redhat-developer/vscode-yaml/issues/565) in case you face vscode errors

**Pull request merged**